websocket-chat/main.go

```diff
@@ -30,8 +30,9 @@ func runHub() {
 				if err := connection.WriteMessage(websocket.TextMessage, []byte(message)); err != nil {
 					log.Println("write error:", err)
 
-					unregister <- connection
-					connection.WriteMessage(websocket.CloseMessage, []byte{})
+					//unregister <- connection
+					//connection.WriteMessage(websocket.CloseMessage, []byte{})
```

Comment:
I think we still need to send close message and close the connection, and then we delete it from map.